### ISSUE TITLE
"Error: All add-ons require a manifest file."

### ISSUE THREAD

**Issue by gkoberger** [mozilla/addons-server#3476]
Figured it out

**Comment by ritikakh** [comment]
Hi Could you please tell what was the issue and what did you figure out? I am struggling with the exactly same error.

**Comment by andymckay** [comment]
The example in this case does not have manifest.json in the root of the zip file, it's at `borderify/manifest.json`.

See: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Publishing_your_WebExtension#1._Zip_up_your_add-on%27s_files for more.

**Comment by ritikakh** [comment]
Hi Andy,

Thank you so much for your reply. I figured out the issue. It was silly on
my part.

I am not sure of the next thing that I am running into. It'll be great if
you could help.

The validation process found these issues that can lead to rejections:

   - Markup should not be passed to `innerHTML` dynamically.

Any kind of guidance would be helpful.

Thanks,
Ritika

Ritika
Graduate Student
Computer Science Department
University of Southern California
<http://www-scf.usc.edu/%7Espmahaja/>

**Comment by ritikakh** [comment]
Hi There,

I have been working with signing an add on from Mozilla. I do have a small
question. In the *install.rdf* file, where we mention the min and max
version, I understand that the min and max version's are the range of the
versions that add-on should support. I mentioned my max version to be *49.**,
but it does work for firefox *50*. Could you please help me understand this
factor.

Thanks for your help.

-Ritika

Ritika
Graduate Student
Computer Science Department
University of Southern California
<http://www-scf.usc.edu/%7Espmahaja/>

**Comment by andymckay** [comment]
We try to use issues to deal with bugs and features on the server. It shouldn't be a support forum, many other places exist for that, please check out:

https://wiki.mozilla.org/Add-ons#Getting_in_touch

Which includes links to a mailing list and discussion forums.

**Comment by ritikakh** [comment]
Hi Andy,

I do have one last question for you. I want to update the max version in my
signed add-on. Is there an option where I can update my install.rdf after
having it installed on Mozilla.

Thanks,
Ritika

Ritika
Graduate Student
Computer Science Department
University of Southern California
<http://www-scf.usc.edu/%7Espmahaja/>

**Comment by ritikakh** [comment]
Also when I try to update the version and try uploading the file to get it
signed, it generates an error saying "Duplicate add on ID found", which is
like obvious but I need a way to update the max version in my file.

Thanks,
Ritika

Ritika
Graduate Student
Computer Science Department
University of Southern California
<http://www-scf.usc.edu/%7Espmahaja/>

**Comment by ritikakh** [comment]
Hi There,

We could update the version, could get the extension signed, but when we
try to install it, it gives the following error:

[image: Inline image 1]

Any kind of help would be appreciated. I did get the new xpi file from
mozilla when I got the extension signed.

I will be looking forward to your kind reply.

Thanks,
Ritika

Ritika
Graduate Student
Computer Science Department
University of Southern California
<http://www-scf.usc.edu/%7Espmahaja/>
